api/v1beta1/rabbitmqcluster_types.go

```diff
@@ -153,10 +149,17 @@ type VaultTLSSpec struct {
 }
 
 func (spec *VaultSpec) TLSEnabled() bool {
-	return spec.TLS.PathCertificate != ""
+	return spec.TLS.PKIIssuerPath != ""
 }
 func (spec *VaultSpec) DefaultUserSecretEnabled() bool {
-	return spec.PathDefaultUser != ""
+	return spec.DefaultUserPath != ""
+}
+func (spec *VaultSpec) ResolveDefaultUserUpdaterInage() string {
+	if spec.DefaultUserUpdaterImage == "" {
+		return "rabbitmqoperator/admin-password-updater:0.1.1"
+	}
+	return spec.DefaultUserUpdaterImage
+
 }
```

Comment:
Why don't we use kubebuilder default?
E.g. as done in https://github.com/rabbitmq/cluster-operator/blob/588649d4cb6c0e8efbd0f1639cd5b89e55363c5f/api/v1beta1/rabbitmqcluster_types.go#L54

Reply:
If we use kubebuilder, it provides that default value under secretBackend.vault which several tests cases did not expect hence they failed.  For this reason, I decided to provide the default via logic rather than thru this kubebuilder annotation.

Reply:
Fixed in https://github.com/rabbitmq/cluster-operator/pull/846/commits/9b9a3d0d61f0ed838b279ac767dcdbbdc236a0b2